Source/UI/Support/UserDefaultsExtension.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+private enum UserDefaultsKey: String, CaseIterable {
+    case hostname = "IPFTestHostname"
+    case port = "IPFTestPort"
+    case configType
+    case streams
+    case duration
+    case enableHaptics
+    case enableSounds
+}
+
+public extension UserDefaults {
+    static var hostname: String {
+        get { read(.hostname) ?? "" }
+        set { store(newValue, for: .hostname) }
+    }
+
+    static var port: UInt {
+        get { read(.port) ?? 5201 }
+        set { store(newValue, for: .port) }
+    }
+
+    static var configType: UInt {
+        get { read(.configType) ?? 1 }
+        set { store(newValue, for: .configType) }
+    }
+
+    static var streams: UInt {
+        get { read(.streams) ?? 5 }
+        set { store(newValue, for: .streams) }
+    }
+
+    static var duration: UInt {
+        get { read(.duration) ?? 10 }
+        set { store(newValue, for: .duration) }
+    }
+
+    static var enableHaptics: Bool {
+        get { read(.enableHaptics) ?? true }
+        set { store(newValue, for: .enableHaptics) }
+    }
+
+    static var enableSounds: Bool {
+        get { read(.enableSounds) ?? true }
+        set { store(newValue, for: .enableSounds) }
+    }
+
+    static func reset() {
+        UserDefaultsKey.allCases.forEach {
+            UserDefaults.standard.removeObject(forKey: $0.rawValue)
+        }
+    }
+}
+
+private extension UserDefaults {
```

Comment:
What if the OS were to override these methods? `read` and `store` are pretty generic names.

Reply:
Didn't have any side effects AFAIK, but renamed to readValue and storeValue.